railgun/test/integration/ingress_bulk_test.go

```diff
@@ -0,0 +1,145 @@
+//+build integration_tests
+
+package integration
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kong/kubernetes-ingress-controller/pkg/annotations"
+	"github.com/kong/kubernetes-testing-framework/pkg/utils/kubernetes/generators"
+)
+
+const (
+	maxBatchSize             = 50
+	testBulkIngressNamespace = "ingress-bulk-testing"
+)
+
+// TestIngressBulk attempts to validate functionality at scale by rapidly deploying a large number of ingress resources.
+func TestIngressBulk(t *testing.T) {
+	ctx := context.Background()
+
+	t.Logf("creating namespace %s for testing", testBulkIngressNamespace)
+	namespace := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testBulkIngressNamespace}}
+	namespace, err := env.Cluster().Client().CoreV1().Namespaces().Create(ctx, namespace, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	defer func() {
+		t.Logf("cleaning up namespace %s", testBulkIngressNamespace)
+		require.NoError(t, env.Cluster().Client().CoreV1().Namespaces().Delete(ctx, namespace.Name, metav1.DeleteOptions{}))
+		require.Eventually(t, func() bool {
+			_, err := env.Cluster().Client().CoreV1().Namespaces().Get(ctx, namespace.Name, metav1.GetOptions{})
+			if err != nil {
+				if errors.IsNotFound(err) {
+					return true
+				}
+			}
+			return false
+		}, ingressWait, waitTick)
+	}()
+
+	t.Log("deploying a minimal HTTP container to be exoposed via ingress")
+	container := generators.NewContainer("httpbin", httpBinImage, 80)
```

Comment:
Another way to remove duplicate code is add container into Deployment spec, and create deployment only.

Reply:
This test is only using the existing tools and facilities provided thus far, but I would argue it is out of scope to add the creation of more generalized testing tools to its scope. I'm not currently seeing a strong reason to make such a distinction, but if you'd like to create a follow up item for that I'm open to it.